apps/linode-marketplace-filecloud/roles/filecloud/tasks/main.yml

```diff
@@ -0,0 +1,96 @@
+# FileCloud installation
+- name: Adding Keys for MongoDB
+  get_url:
+      url: https://pgp.mongodb.com/server-6.0.asc
+      dest: /usr/share/keyrings/mongodb-server-6.0.asc
+      mode: '0644'
+      force: true 
+
+- name: adding mongodb repo
+  apt_repository:
+    repo: "deb [ arch=amd64,arm64 signed-by=/usr/share/keyrings/mongodb-server-6.0.asc ] https://repo.mongodb.org/apt/ubuntu jammy/mongodb-org/6.0 multiverse"
+    state: present
+
+- name: Adding Keys for FileCloud
+  get_url:
+      url: https://repo.filecloudlabs.com/static/pgp/filecloud.asc
+      dest: /usr/share/keyrings/filecloud.asc
+      mode: '0644'
+      force: true
+
+- name: adding filecloud repo
+  apt_repository:
+    repo: "deb [ arch=amd64 signed-by=/usr/share/keyrings/filecloud.asc ] https://repo.filecloudlabs.com/apt/ubuntu jammy/filecloud/23.232 main"
+    state: present
+
+- name: running apt-get update
+  apt:
+    update_cache: yes
+    upgrade: yes
+
+- name: Installing the Apache and MongoDB servers.
+  apt:
+    name:
+    - apache2
+    - mongodb-org
+    - pigz
+    state: present
+
+- name: Installing PHP 8.2 and dependencies
+  apt:
+    name: php8.2*
+    install_recommends: no
+    state: present
+
+- name: Installing FileCloud! :)
+  apt:
+    name: filecloud
+    state: present
+
+- name: grabbing private ip
+  shell: 
+    cmd: hostname -i
+  register: private_ip
```

Comment:
This needs to be reworked.  If we need a private_ip variable, it must be set in the provision.yml so that it can be written to the `group_vars/linode/vars` and then called later. Avoid the use of shell where possible.